### PR TITLE
Update rubygem-foreman_remote_execution to 1.8.3

### PR DESCRIPTION
**Pull request by iNecas**
Adds rubygem-foreman_remote_execution-cockpit subpackage with the cockpit integration bits and pieces

For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.21
* [ ] 1.20

See Foreman's [plugin maintainer documentation](https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

File: packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec [hunk 1]
```diff
@@ -72,19 +85,33 @@ This package contains documentation for rubygem-%{gem_name}.
...
+
 %posttrans
 %{foreman_db_migrate}
 %{foreman_db_seed}
 %{foreman_apipie_cache}
 %{foreman_restart}
 exit 0
 
+%post cockpit
+%systemd_postun_with_restart foreman-cockpit.service
```
Comment (ekohl) [inline packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec]: Why is this here?

File: packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec [hunk 2]
```diff
@@ -72,19 +85,33 @@ This package contains documentation for rubygem-%{gem_name}.
...
+%systemd_post foreman-cockpit.service
+
+%postun cockpit
+%systemd_postun_with_restart foreman-cockpit.service
+%systemd_post foreman-cockpit.service
```
Comment (ekohl) [inline packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec]: This also surprises me

File: packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec [hunk 3]
```diff
@@ -53,6 +54,18 @@ Provides: foreman-plugin-remote_execution
 A plugin bringing remote execution to the Foreman, completing the config
 management functionality with remote management functionality
 
+%package cockpit
+BuildArch:  noarch
+Requires:   cockpit
+Requires(post): systemd-sysv
```
Comment (ekohl) [inline packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec]: I wonder why this is needed. I thought it's only for legacy sysv init scripts but this is a native systemd service.

File: packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec (hunk 1 above)
Comment (iNecas) [reply packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec]: Inspired heavily by https://github.com/theforeman/foreman-packaging/blob/rpm/develop/packages/foreman/foreman/foreman.spec#L1031

Expands to:

```
systemctl daemon-reload >/dev/null 2>&1 || : 
if [ $1 -ge 1 ] ; then 
        # Package upgrade, not uninstall 
        systemctl try-restart foreman-cockpit.service >/dev/null 2>&1 || : 
fi 


if [ $1 -eq 1 ] ; then 
        # Initial installation 
        systemctl preset foreman-cockpit.service >/dev/null 2>&1 || : 
fi 
```

Looks useful

File: packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec (hunk 2 above)
Comment (iNecas) [reply packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec]: Ditto

File: packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec (hunk 3 above)
Comment (iNecas) [reply packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec]: Because we have it in foreman.spec: https://github.com/theforeman/foreman-packaging/blob/rpm/develop/packages/foreman/foreman/foreman.spec#L52 seems like it should not be there as well. Updated this spec

**Comment by iNecas** [comment]
The `%preun` macros added

File: packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec (hunk 1 above)
Comment (ekohl) [reply packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec]: I think that one in foreman.spec is already incorrect and we should rather be using `%systemd_post` in `%post`.

File: packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec (hunk 2 above)
Comment (ekohl) [reply packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec]: The `%systemd_postun_with_restart` should be sufficient here.

File: packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec (hunk 1 above)
Comment (iNecas) [reply packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec]: Updated

File: packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec (hunk 2 above)
Comment (iNecas) [reply packages/plugins/rubygem-foreman_remote_execution/rubygem-foreman_remote_execution.spec]: Updated

**Comment by ehelms** [comment]
[test rpm]
